src/components/search/SearchDirective.js

```diff
@@ -1,756 +1,259 @@
-            // number of lists
-            scope.nbOfSuggestionsLists = {
-              'ga-search-1': false,
-              'ga-search-2': false,
-              'ga-search-3': false
-            };
+  module.directive('gaSearch',
+      function($timeout) {
+        return {
+          restrict: 'A',
+          replace: true,
+          templateUrl: 'components/search/partials/search.html',
+          controller: 'GaSearchDirectiveController',
+          scope: {
+            options: '=gaSearchOptions',
+            map: '=gaSearchMap',
+            searchFocused: '=gaSearchFocused'
+          },
+          link: function($scope, element, attrs) {
+            $scope.input = element.find('input');
 
-            var setListCount = function() {
-              var counter = 0;
-              counter = !hasLocationResults ? counter : counter + 1;
-              counter = !hasFeatureResults ? counter : counter + 1;
-              counter = !scope.hasLayerResults ? counter : counter + 1;
-              for (var cssClass in scope.nbOfSuggestionsLists) {
-                if (cssClass === 'ga-search-' + counter) {
-                  scope.nbOfSuggestionsLists[cssClass] = true;
-                } else {
-                  scope.nbOfSuggestionsLists[cssClass] = false;
+            $scope.key = function(evt) {
+              if (evt.keyCode == 13) {
+                if (evt.target && evt.target.blur) {
+                  evt.target.blur();
                 }
               }
             };
+            // Result set announces their results (put here because we need
+            // element)
+            $scope.childoptions.announceResults = function(type, nr) {
+              $scope.restat.update(type, nr);
 
-            viewDropDown.on('gaSuggestionsRendered', function(evt) {
-              var el;
-              if (viewDropDown.isVisible()) {
-                $timeout(setListCount);
-                el = element.find('.tt-dataset-' + evt.data.name);
-                el.attr('ng-class', 'nbOfSuggestionsLists');
-                $compile(el)(scope);
-                if (el) {
-                  el = el.find('.tt-suggestions');
-                  if (el) {
-                    gaSwisssearch.feed(el);
-                  }
-                  el.scrollTop(0);
-                }
-              }
-              gaSwisssearch.check();
-            });
-
-            scope.clearInput = function() {
-              taElt.val('');
-              taElt.data('ttView').inputView.setQuery('');
-              scope.query = '';
-              gaMarkerOverlay.remove(map);
-              unregisterMove();
-              viewDropDown.clearSuggestions();
-              scope.searchFocused = false;
-              // Display the placeholder
-              if (gaBrowserSniffer.msie < 10) {
+              if (swisssearchActive &&
+                  $scope.restat.done()) {
                 $timeout(function() {
```

Comment:
Yes. The digest cycle is needed to acutally render the results. Otherwise, we won't find .ga-search-item elements.